### DISCUSSION TITLE
Running docker compose run as action step no longer outputs logs

### DISCUSSION THREAD

**Discussion by ste26054** [community/community#11011] [Actions] [answered]
Hi,

I have recently encountered a regression where part of our workflow has a step which runs a `docker compose run` command.
This command no longer outputs the container logs in the job logs view.

A quick fix is to run instead `docker-compose run` (note the added `-`), but then we loose syntactic coloration 😕  

I have opened an issue in the [Github Actions Runner repository #1627](https://github.com/actions/runner/issues/1627) if you wish to have more details / screenshots on how to reproduce (I was then advised to post the issue here since this is not on the runner side).

Thanks for your help.

**Comment by jzeimen** [comment]
I think this is still a real issue. The `docker-compose` workaround isn't working when you need to use buildkit. Even with `DOCKER_BUILDKIT: 1` in `env`. I'm running on `ubuntu-latest`. On Mac `docker-compose` or `docker compose` both work to build/run but in the runner docker-compose doesn't run. Unless there is another workaround I don't see a way to get the output and use buildkit with docker compose.

**Comment by ethomson** [comment]
Thanks @ste26054, I moved your issue to the appropriate repository for the actions team to take a look.

**Comment by ste26054** [comment, marked as answer]
For those coming here, due to a change in docker compose `2.2.3`, it is now needed to add the option `--no-TTY` to `docker compose run`. Actions will then show the compose output log as previously expected.
